scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java

```diff
@@ -1018,17 +1018,16 @@ private void endJob(JobData jobData, TerminationData terminationData, InternalTa
             }
         }
 
+        Set<TaskId> tasksToUpdate = null;
         // if job has been killed
         if (jobStatus == JobStatus.KILLED) {
-            Set<TaskId> tasksToUpdate = job.failed(null, jobStatus);
-            dbManager.updateAfterJobKilled(job, tasksToUpdate);
-            updateTasksInSchedulerState(job, tasksToUpdate);
+            tasksToUpdate = job.failed(null, jobStatus);
 
         } else {
             // don't tamper the original job status if it's already in a
             // finished state (failed/canceled)
             if (jobStatus != JobStatus.FINISHED) {
-                Set<TaskId> tasksToUpdate = job.failed(task.getId(), jobStatus);
+                tasksToUpdate = job.failed(task.getId(), jobStatus);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "task" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)